blockonomics.py

```diff
@@ -1,32 +1,181 @@
 import requests
 import configloader
+import logging
+import sqlalchemy
+import datetime
+from decimal import Decimal
+import json
+import re
+
+import database as db
+
+log = logging.getLogger(__name__)
 
 # Define all the database tables using the sqlalchemy declarative base
 class Blockonomics:
+
+    @staticmethod
     def fetch_new_btc_price():
         url = 'https://www.blockonomics.co/api/price'
         params = {'currency':configloader.user_cfg["Payments"]["currency"]}
         r = requests.get(url,params)
         if r.status_code == 200:
           price = r.json()['price']
-          print ('Bitcoin price ' + str(price))
+          log.debug("BTC Price %s" % price)
           return price
         else:
-          print(r.status_code, r.text)
+          log.error("Fetch BTC Price Failed, Status: %s, Response: %s" % (r.status_code, r.content))
+    
+    @staticmethod
+    def _get_secret():
+        return configloader.user_cfg["Bitcoin"]["secret"]
 
+    @staticmethod
     def new_address(reset=False):
         api_key = configloader.user_cfg["Bitcoin"]["api_key"]
-        secret = configloader.user_cfg["Bitcoin"]["secret"]
         url = 'https://www.blockonomics.co/api/new_address'
+        params = {}
         if reset == True:
-          url += '?match_callback='+secret+'&reset=1'
-        else:
-          url += "?match_callback=" + secret
+          params['reset'] = 1
+        
         headers = {'Authorization': "Bearer " + api_key}
-        print(url)
-        r = requests.post(url, headers=headers)
+        r = requests.post(url, headers=headers, params={**params, "match_callback": Blockonomics._get_secret()})
```

Comment:
Multiple stores are currently hard to setup as a callback URL is required.
I have created a separate issue to handle this: https://github.com/DarrenWestwood/greed/issues/54